File: JS/rental-app-js/services/itemService.js
const Item = require('../models/Item');

const items = [
  new Item('Bike'),
  new Item('Camera'),
  new Item('Tent')
];

function getItems() {
  return items;
}

function findByName(name) {
  return items.find(item => item.name.toLowerCase() === name.toLowerCase());
}

module.exports = { getItems, findByName };

File: JAVA/rental-app-java/src/services/ItemService.java
package services;

import models.Item;

import java.util.Arrays;
import java.util.List;

public class ItemService {
    private final List<Item> items = Arrays.asList(
            new Item("Bike"),
            new Item("Camera"),
            new Item("Tent")
    );

    public List<Item> getItems() {
        return items;
    }

    public Item findByName(String name) {
        return items.stream()
                .filter(item -> item.getName().equalsIgnoreCase(name))
                .findFirst().orElse(null);
    }
}
